stubs/Pillow/PIL/ImageFilter.pyi

```diff
@@ -1,115 +1,136 @@
     def __init__(self, size: int = ...) -> None: ...
-    def filter(self, image): ...
+    def filter(self, image) -> Image: ...
 
 class GaussianBlur(MultibandFilter):
     name: str
-    radius: Any
-    def __init__(self, radius: int = ...) -> None: ...
-    def filter(self, image): ...
+    radius: float
+    def __init__(self, radius: float = ...) -> None: ...
+    def filter(self, image) -> Image: ...
 
 class BoxBlur(MultibandFilter):
     name: str
-    radius: Any
-    def __init__(self, radius) -> None: ...
-    def filter(self, image): ...
+    radius: float
+    def __init__(self, radius: float) -> None: ...
+    def filter(self, image) -> Image: ...
 
 class UnsharpMask(MultibandFilter):
     name: str
-    radius: Any
-    percent: Any
-    threshold: Any
-    def __init__(self, radius: int = ..., percent: int = ..., threshold: int = ...) -> None: ...
-    def filter(self, image): ...
+    radius: float
+    percent: int
+    threshold: int
+    def __init__(self, radius: float = ..., percent: int = ..., threshold: int = ...) -> None: ...
+    def filter(self, image) -> Image: ...
 
 class BLUR(BuiltinFilter):
     name: str
-    filterargs: Any
+    filterargs: _FilterArgs
 
 class CONTOUR(BuiltinFilter):
     name: str
-    filterargs: Any
+    filterargs: _FilterArgs
 
 class DETAIL(BuiltinFilter):
     name: str
-    filterargs: Any
+    filterargs: _FilterArgs
 
 class EDGE_ENHANCE(BuiltinFilter):
     name: str
-    filterargs: Any
+    filterargs: _FilterArgs
 
 class EDGE_ENHANCE_MORE(BuiltinFilter):
     name: str
-    filterargs: Any
+    filterargs: _FilterArgs
 
 class EMBOSS(BuiltinFilter):
     name: str
-    filterargs: Any
+    filterargs: _FilterArgs
 
 class FIND_EDGES(BuiltinFilter):
     name: str
-    filterargs: Any
+    filterargs: _FilterArgs
 
 class SHARPEN(BuiltinFilter):
     name: str
-    filterargs: Any
+    filterargs: _FilterArgs
 
 class SMOOTH(BuiltinFilter):
     name: str
-    filterargs: Any
+    filterargs: _FilterArgs
 
 class SMOOTH_MORE(BuiltinFilter):
     name: str
-    filterargs: Any
+    filterargs: _FilterArgs
 
 class Color3DLUT(MultibandFilter):
     name: str
-    size: Any
-    channels: Any
-    mode: Any
+    size: int | tuple[int, int, int]
+    channels: int
+    mode: str | None
     table: Any
-    def __init__(self, size, table, channels: int = ..., target_mode: Any | None = ..., **kwargs) -> None: ...
+    def __init__(
+        self, size: int | tuple[int, int, int], table, channels: int = ..., target_mode: str | None = ..., **kwargs
+    ) -> None: ...
     @classmethod
-    def generate(cls, size, callback, channels: int = ..., target_mode: Any | None = ...): ...
-    def transform(self, callback, with_normals: bool = ..., channels: Any | None = ..., target_mode: Any | None = ...): ...
-    def filter(self, image): ...
+    def generate(
+        cls: Type[Self],
+        size: int | tuple[int, int, int],
+        callback: Callable[[int, int, int], float],
```

Comment:
This gets called with floats and should return an iterable with `channels` elements (a tuple according to the docstring).
```suggestion
        callback: Callable[[float, float, float], Iterable[float]],
```